tests/components/UploadFileDialog/UploadFileDialog.spec.tsx

```diff
@@ -40,20 +40,16 @@ describe("Tests for UploadFileDialog", () => {
       wrapper,
       addFiles() {
         const instance = wrapper.instance() as UploadFileDialog;
-        const root = instance.root.getModel();
-        const fileA = root.newFile("fileA", FileType.JavaScript);
-        const fileB = root.newFile("fileB", FileType.JavaScript);
-        instance.root = ModelRef.getRef(root);
+        const fileA = instance.root.newFile("fileA", FileType.JavaScript);
+        const fileB = instance.root.newFile("fileB", FileType.JavaScript);
         wrapper.update();
         return [fileA, fileB];
       },
       addDirectory(name) {
         const instance = wrapper.instance() as UploadFileDialog;
-        const root = instance.root.getModel();
-        const directory = root.newDirectory(name);
+        const directory = instance.root.newDirectory(name);
```

Comment:
ModelRef.getRef and getModel were introduced to "track" direct mutations to the model. Looks like the `const directory = root.newDirectory(name);` code did just that. I recommend to extract this mutation from the component (and continue using model refs until we weed out all such instances)